packages/core/util/layouts/PrecomputedLayout.ts

```diff
@@ -1,20 +1,31 @@
 import { objectFromEntries } from '..'
-import { RectTuple } from './BaseLayout'
+import {
+  RectTuple,
+  SerializedLayout,
+  BaseLayout,
+  Rectangle,
+} from './BaseLayout'
 
-export default class PrecomputedLayout {
+export default class PrecomputedLayout<T> implements BaseLayout<T> {
   private rectangles: Map<string, RectTuple>
 
   private totalHeight: number
 
-  constructor({ rectangles = {}, totalHeight = 0 }) {
+  public maxHeightReached: boolean
+
+  constructor({ rectangles, totalHeight, maxHeightReached }: SerializedLayout) {
     this.rectangles = new Map(Object.entries(rectangles))
     // rectangles is of the form "featureId": [leftPx, topPx, rightPx, bottomPx]
     this.totalHeight = totalHeight
+    this.maxHeightReached = maxHeightReached
   }
 
-  addRect(id: string): number {
+  addRect(id: string) {
     const rect = this.rectangles.get(id)
     if (!rect) {
+      if (this.maxHeightReached) {
```

Comment:
still curious about this extra check on this.maxHeightReached, any context that triggers this?